source/module_elecstate/magnetism.cpp

```diff
@@ -5,14 +5,10 @@
 
 Magnetism::Magnetism()
 {
-    this->tot_magnetization = 0.0;
-    this->abs_magnetization = 0.0;
-    this->start_magnetization = nullptr;
 }
 
 Magnetism::~Magnetism()
 {
-    delete[] this->start_magnetization;
```

Comment:
what have you done to 'new start_magnetization'?

Reply:
The actual calculation of magnetic data is done in Atom; here it is just used as an intermediate variable to store data.

Reply:
In the setup function of unitcell, memory space of start_magnetization is allocated, and in Magnetism, memory space is released. In the program, data is written only when reading atoms in read_atoms(unitcell::setup), and then saved to atoms[it].mag[ia]. Subsequently, atom[it].mag[ia] is always called. This array is duplicated.It will be refactored in the future.